pkg/apis/syn/v1alpha1/tenant_types_test.go

```diff
@@ -82,3 +84,221 @@ func TestTenant_ApplyTemplate(t *testing.T) {
+					ResourceNames: []string{"other-resource"},
+				},
+			},
+		},
+		updated: true,
+		assert: func(t *testing.T, role *rbacv1.Role) {
+			require.Len(t, role.Rules, 1)
+			assert.Len(t, role.Rules[0].ResourceNames, 2)
+			assert.Contains(t, role.Rules[0].ResourceNames, "my-tenant")
+			assert.Contains(t, role.Rules[0].ResourceNames, "other-resource")
+		},
+	},
+	"add new rule when no rule matches": {
+		tenant: &synv1alpha1.Tenant{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "my-tenant",
+				Namespace: "my-namespace",
+			},
+		},
+		role: &rbacv1.Role{
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{"some-group"},
+					Verbs:     []string{"update"},
+					Resources: []string{"some-resource"},
+					ResourceNames: []string{"before"},
+				},
+			},
+		},
+		updated: true,
+		assert: func(t *testing.T, role *rbacv1.Role) {
+			require.Len(t, role.Rules, 2)
+			assert.Len(t, role.Rules[1].ResourceNames, 1)
+			assert.Contains(t, role.Rules[1].ResourceNames, "my-tenant")
+		},
+	},
+	"the correct rule gets updated": {
+		tenant: &synv1alpha1.Tenant{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "my-tenant",
+				Namespace: "my-namespace",
+			},
+		},
+		role: &rbacv1.Role{
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{"some-group"},
+					Verbs:     []string{"update"},
+					Resources: []string{"some-resource"},
+					ResourceNames: []string{"before"},
+				},
+				{
+					APIGroups: []string{synv1alpha1.SchemeGroupVersion.Group},
+					Verbs:     []string{"get"},
+					Resources: []string{"tenants", "clusters"},
+					ResourceNames: []string{"other-resource"},
+				},
+				{
+					APIGroups: []string{"some-other-group"},
+					Verbs:     []string{"delete"},
+					Resources: []string{"some-other-resource"},
+					ResourceNames: []string{"after"},
+				},
+			},
+		},
+		updated: true,
+		assert: func(t *testing.T, role *rbacv1.Role) {
+			require.Len(t, role.Rules, 3)
+			assert.Len(t, role.Rules[1].ResourceNames, 2)
+			assert.Contains(t, role.Rules[1].ResourceNames, "my-tenant")
+			assert.Contains(t, role.Rules[1].ResourceNames, "other-resource")
+		},
+	},
+}
+
+func TestTenant_SynchronizeRole(t *testing.T) {
+	for name, tt := range tenantUpdateRoleTests {
+		t.Run(name, func(t *testing.T) {
+			assert.Equal(t, tt.updated, tt.tenant.SynchronizeRole(tt.role))
+			tt.assert(t, tt.role)
+		})
+	}
+}
+
+func assertManagedByLabel(t *testing.T, obj v1.ObjectMeta, value string) {
+	v, ok := obj.Labels["app.kubernetes.io/managed-by"]
+	assert.True(t, ok, "the label `app.kubernetes.io/managed-by` is not set")
+	assert.Equal(t, value, v)
+}
+
+func assertOwnerReference(t *testing.T, obj v1.Object, owner *synv1alpha1.Tenant) {
+	gvk := synv1alpha1.SchemeGroupVersion.WithKind(owner.Kind)
+	refs := obj.GetOwnerReferences()
+	require.Len(t, refs, 1)
+	assert.Equal(t, gvk.GroupVersion().String(), refs[0].APIVersion)
+	assert.Equal(t, gvk.Kind, refs[0].Kind)
+	assert.Equal(t, owner.GetName(), refs[0].Name)
```

Comment:
When using the utility function from controller-runtime, asserting all properties may be overkill
```suggestion
	refs := obj.GetOwnerReferences()
	require.Len(t, refs, 1)
	assert.Equal(t, owner.GetName(), refs[0].Name)
```